### ISSUE TITLE
feat: Overlay component should yield enough Information Upon Opening with Screen Readers

### ISSUE THREAD

**Issue by gurpreetonebcg** [ionic-team/ionic-framework#23602]
# Feature Request

**Ionic version:**

```
@ionic/angular 5.6.0
```

**Describe the Feature Request**

When double tapping upon the button with Talkback or Voiceover, the dialogue should be designed as such that it can be viewable in text form and read out loud by Talkback and voiceover. The modals should also remain in focus until pressing the ‘cancel’ button or the ‘okay’ button to confirm the action. It should announce the label when opening so that user can identify that what popup is opened on screen. This should be on every overlay component like
- Alerts
- Action sheet
- Modal
- Date-time picker

**Related Code**

[Downlead code here](https://drive.google.com/file/d/1TfR-9kxiYYeVQkr6WOEjHaSuN_ZIq9my/view?usp=sharing) 

**Additional Context**

```
Ionic:

   Ionic CLI                     : 6.11.8
   Ionic Framework               : @ionic/angular 5.6.0
   @angular-devkit/build-angular : 0.1101.4
   @angular-devkit/schematics    : 11.1.4
   @angular/cli                  : 11.1.4
   @ionic/angular-toolkit        : 3.1.0

Cordova:

   Cordova CLI       : 10.0.0
   Cordova Platforms : android 9.0.0
   Cordova Plugins   : cordova-plugin-ionic-keyboard 2.2.0, cordova-plugin-ionic-webview 4.2.1, (and 4 other plugins)

Utility:

   cordova-res : 0.11.0
   native-run  : 1.0.0
```

**Comment by liamdebeasi** [comment]
Thanks for the issue. Can you please provide some steps to reproduce the issue? I am seeing that the focus is being placed inside of the overlays properly and that VoiceOver is announcing the title of the overlays.

**Comment by gurpreetonebcg** [comment]
Thanks for reply, Please try this on any Action Sheet or Date Time picker, you will see it only announces as "web dialog"

**Comment by liamdebeasi** [comment]
I am testing this on a device running iOS 14.6 and VoiceOver does announce "web dialog", but then it moves focus to the title in the action sheet and reads "Albums".

Can you please describe your expected behavior?

**Comment by gurpreetonebcg** [comment]
Am testing this in android, please open any date time picker modal, you will notice there is no indication that which modal is opened. Please refer to the attached video. https://drive.google.com/file/d/1GVP__IrPTsLz6fIZ9l8TdYxFTRQW1xbw/view?usp=sharing

**Comment by liamdebeasi** [comment]
Thanks for the issue and apologies for the delay. There have been numerous improvements made to overlay components in Ionic that should resolve this issue. I recommend updating to Ionic v6.3.4 or newer to receive these fixes. I am going to close this for now, but please let me know if you have any additional questions.
